clients/client-core/src/client/real_messages_control/real_traffic_stream.rs

```diff
@@ -222,8 +241,199 @@ where
     /// Accessor to the common instance of network topology.
     topology_access: TopologyAccessor,
 
-    /// Buffer containing all real messages received. It is first exhausted before more are pulled.
-    received_buffer: VecDeque<RealMessage>,
+    /// Buffer containing all real messages keyed by transmission lane.
+    received_buffer: ReceivedBuffer,
+
+    /// Incoming channel for being notified of closed connections, to avoid sending traffic
+    /// unnecessary
+    closed_connection_rx: ClosedConnectionReceiver,
+}
+
+#[derive(Default)]
+struct ReceivedBuffer {
```

Comment:
While the name works in that context, I'd perhaps rename it because we already have `ReceivedBuffer` and all of that for messages received from mixnet, Uhm... maybe something like `IncomingBuffer` or `PendingBuffer` or `TransmissionBuffer`? I'm not sure honestly

Reply:
`TransmissionBuffer` sounds like a good one